### PR TITLE
change(ios/app): Generate offline help from markdown

### PR DESCRIPTION
**Pull request by darcywong00**
This PR
* checks in the markdown help from keymanapp/help.keyman.com#315 
* updates build-help.sh to generate offline help from the markdown files.
* moved htmlink.lua to resources/build since it's now shared between windows, android, ios help

@jahorton - can you finish these TODOs since I don't have XCode

TODO
- [x] Update kmbuild.sh to call `build-help.sh html` to generate the offline help as part of the build process
- [x] Check if the offline help appears

### CODE REVIEW

**Comment by jahorton** [comment]
A note for later:  the help-related view code uses `UIWebView` instead of `WKWebView`.  Unfortunately, only the latter supports forward and back navigation, which would be _really_ handy with the offline help.  Again unfortunately, `WKWebView` is only supported within XIB files (currently used by help) for iOS 11.0+.

As a result, once a user's drilled down to a specific help page, they can't navigate back to the index; they have to close & reopen help.  Also, I noticed that the welcome page's link didn't properly connect to the start-section for some reason.

Can't wait for us to boost the minimum iOS version after 14.0; there are all sorts of before-and-after 11.0 checks in the code that would be great to clean up.  🤞

**Comment by jahorton** [comment]
One thing I've been wondering with these PRs - is there no way to download the help.keyman.com CSS file and have `pandoc` link that in for the offline help?

It's not a blocker, but it seems like something that should be possible.

**Comment by jahorton** [comment]
I think it's good to go, but I'm not sure I should be reviewing, as I contributed a non-insignificant portion of the code.

**Comment by mcdurdin** [comment]
> is there no way to download the help.keyman.com CSS file and have `pandoc` link that in for the offline help?

The help.keyman.com CSS is not a great match (but we could use a well trimmed version of it); certainly it's possible to specify CSS for pandoc and I suggest we work on that once we have all the help migrated.

**Comment by darcywong00** [comment]
> but I'm not sure I should be reviewing, as I contributed a non-insignificant portion of the code.

fwiw, the Swift changes lgtm

File: ios/keyman/Keyman/Keyman/InfoViewController/InfoViewController.swift [hunk 1]
```diff
@@ -43,27 +43,29 @@ class InfoViewController: UIViewController, UIWebViewDelegate {
...
-    let filePath = offlineHelpBundle.path(forResource: "index.php", ofType: "html", inDirectory: nil)
+    // Safari won't recognize the contents without the .html ending.
+    let filePath = offlineHelpBundle.path(forResource: "index", ofType: "html", inDirectory: nil)
     webView.loadRequest(URLRequest(url: URL.init(fileURLWithPath: filePath!)))
   }
 
   private func loadFromServer() {
     let appVersion = Version.current.majorMinor
-    let url = "https://help.keyman.com/products/iphone-and-ipad/\(appVersion.plainString)/?embed=ios"
+    let url =  "\(KeymanHosts.HELP_KEYMAN_COM)/products/iphone-and-ipad/\(appVersion.plainString)/"
```
Comment (jahorton) [inline ios/keyman/Keyman/Keyman/InfoViewController/InfoViewController.swift:68]: I'm a bit sad at the loss of `/?embed=ios` - a lot of the desktop-site nav filler is back as a result.  But, the new markdown-based pages don't support it, so... oh well.

**Review by jahorton — APPROVED** [review]
Well, it sounds like between my review of the documentation + help batching scripts and @darcywong00 's review of the iOS script + Swift changes, we can combine the two half-reviews for a full LGTM?

File: ios/keyman/Keyman/Keyman/InfoViewController/InfoViewController.swift (hunk 1 above)
Comment (mcdurdin) [reply ios/keyman/Keyman/Keyman/InfoViewController/InfoViewController.swift:68]: I guess we could leave `?embed=ios` in as we can always add that support in later if we want.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.243-beta](https://keyman.com/downloads/releases/beta/14.0.243)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.17-alpha](https://keyman.com/downloads/releases/alpha/15.0.17)

**Pull request merged**